### PR TITLE
feat: hold onto adminFacets in bootstrap

### LINKED ISSUE #6034: bootstrap does not save adminFacets from startInstance etc.
## Describe the bug

I was looking at some code with @Chris-Hibbert  `const { creatorFacet } = await E(zoe).startInstance(...)`

(meeting time.  IOU more)

## Platform Environment
 - what OS are you using? what version of Node.js?
 - is there anything special/unusual about your platform?
 - what version of the Agoric-SDK are you using? (run `git describe --tags --always`)

## Screenshots
If applicable, add screenshots to help explain your problem, especially for UI interactions.

### PR DESCRIPTION
**Pull request by Chris-Hibbert**
closes: #6034

## Description

Save the AdminFacets from the core economy contracts during bootstrap.

This saves them in spaces, so they can be used by big-hammer scripts.

This makes all the contracts started in bootstrap use a consistent format for saving their public, private, governance and admin facets. Rather than all being top level names, each contract has a single top-level name, like `ammFacets`, which contains `{ creatorFacet, governorCreatorFacet, adminFacet, publicFacet }`, and sometimes an `instance`.

`feeDistributor` and `psmCharter` are not governed, so they don't have `adminFacet`s, and `psmCharter` has no `publicFacet`.

### Security Considerations

Wouldn't want to misplace these.

### Documentation Considerations

None.

### Testing Considerations

Untested.

### CODE REVIEW

**Comment by dckc** [comment]
@Chris-Hibbert let me know if getting the types updated is as tricky as I suspect it might be. They mostly live in `packages/vats/src/core/types.js`.

File: packages/inter-protocol/src/proposals/econ-behaviors.js [hunk 1]
```diff
@@ -46,28 +46,43 @@ const sanitizePathSegment = name => {
...
+ */
+
 /**
  * @typedef { WellKnownSpaces & ChainBootstrapSpace & EconomyBootstrapSpace
  * } EconomyBootstrapPowers
  * @typedef {PromiseSpaceOf<{
  *   ammInstanceWithoutReserve: Instance,
  *   ammCreatorFacet: XYKAMMCreatorFacet,
  *   ammGovernorCreatorFacet: GovernedContractFacetAccess<XYKAMMPublicFacet,XYKAMMCreatorFacet>,
+ *   ammGovernorAdminFacet: AdminFacet,
```
Comment (dckc) [inline packages/inter-protocol/src/proposals/econ-behaviors.js]: hm. I suppose this works, but what I had in mind was to capture the whole result of `E(zoe).startInstance(ammGovernorInstallation)` in one field, replacing `ammGovernorCreatorFacet`.

File: packages/inter-protocol/src/proposals/econ-behaviors.js [hunk 2]
```diff
@@ -46,28 +46,43 @@ const sanitizePathSegment = name => {
...
  *   economicCommitteeCreatorFacet: CommitteeElectorateCreatorFacet,
  *   feeDistributorCreatorFacet: import('../feeDistributor.js').FeeDistributorCreatorFacet,
  *   feeDistributorPublicFacet: import('../feeDistributor.js').FeeDistributorPublicFacet,
```
Comment (dckc) [inline packages/inter-protocol/src/proposals/econ-behaviors.js]: all 3 feeDistributor facets would fit in one neat package :)

**Review by dckc — COMMENTED** [review]
This is good progress, but there are relevant calls to `startInstance()` in `packages/vats` too; for example, in `addBankAssets()` in `packages/vats/src/core/basic-behaviors.js`.

There's one in `mintInitialSupply` too... though that's a short-lived contract by design. Let's see... that TODO about shutting it down is done in `packages/vats/src/centralSupply.js`, I see.

File: packages/inter-protocol/src/proposals/econ-behaviors.js [hunk 3]
```diff
@@ -500,22 +519,25 @@ export const startVaultFactory = async (
...
-          initialShortfallInvitation,
-          marshaller,
-          storageNode,
-        },
-      }),
-    );
+  const {
+    creatorFacet: governorCreatorFacet,
+    instance: governorInstance,
+    adminFacet,
```
Comment (dtribble) [inline packages/inter-protocol/src/proposals/econ-behaviors.js:461]: Hwo will this style work with multiple PSMs?

Comment (dckc) [reply packages/inter-protocol/src/proposals/econ-behaviors.js:461]: It won't. hence #6146

File: packages/inter-protocol/src/proposals/addAssetToVault.js [hunk 4]
```diff
@@ -320,9 +321,7 @@ export const getManifestForAddAssetToVault = (
       },
       [addAssetToVault.name]: {
         consume: {
-          vaultFactoryCreator: true,
-          ammCreatorFacet: true,
-          reserveCreatorFacet: true,
+          vaultFactoryFacets: true,
```
Comment (dckc) [inline packages/inter-protocol/src/proposals/addAssetToVault.js]: :+1:

File: packages/inter-protocol/src/proposals/econ-behaviors.js (hunk 2 above)
Comment (Chris-Hibbert) [reply packages/inter-protocol/src/proposals/econ-behaviors.js]: Done.

File: packages/inter-protocol/src/proposals/econ-behaviors.js (hunk 1 above)
Comment (Chris-Hibbert) [reply packages/inter-protocol/src/proposals/econ-behaviors.js]: Pervasively done.

**Review by turadg — APPROVED** [review]
👍 to organizing. I have some hesitation to repeating 'Facets' and wonder if we could drop that from the grouping without confusing things. E.g. that `vaultFactoryFacets.governorCreatorFacet` be simply `vaultFactory.governorCreatorFacet`.

For the git history "save" made me think these objects were going into durable storage. Upon reading the code I see they're just being held in the space. Maybe "hold" would be less at risk of confusion.

File: packages/vats/src/core/startWalletFactory.js [hunk 5]
```diff
@@ -146,12 +146,7 @@ export const startWalletFactory = async (
       consume: { [Stable.symbol]: feeIssuerP },
     },
   },
-  {
-    options: {
-      perAccountInitialValue = (StableUnit * 25n) / 100n,
-      walletBridgeId = BRIDGE_ID.PROVISION_SMART_WALLET,
```
Comment (dckc) [inline packages/vats/src/core/startWalletFactory.js]: are you sure this should be removed?

**Review by dckc — CHANGES_REQUESTED** [review]
removing the `walletBridgeId` setting looks wrong

File: packages/vats/src/core/startWalletFactory.js (hunk 5 above)
Comment (Chris-Hibbert) [reply packages/vats/src/core/startWalletFactory.js]: lint complained that walletBridgeId is unused. How do I make it happy without removing it?

File: packages/vats/src/core/startWalletFactory.js [hunk 6]
```diff
@@ -238,8 +236,8 @@ export const startWalletFactory = async (
   });
 
   await Promise.all([
-    E(bridgeManager).register(walletBridgeId, handler),
-    E(E.get(econCharterStartResult).creatorFacet).addInstance(
+    E(bridgeManager).register(BRIDGE_ID.PROVISION, handler),
```
Comment (dckc) [inline packages/vats/src/core/startWalletFactory.js]: > How do I make it happy ...?

undo this change too; put it back to `walletBridgeId`.

File: packages/vats/src/core/startWalletFactory.js (hunk 5 above)
Comment (Chris-Hibbert) [reply packages/vats/src/core/startWalletFactory.js]: reverted

File: packages/vats/src/core/startWalletFactory.js (hunk 6 above)
Comment (Chris-Hibbert) [reply packages/vats/src/core/startWalletFactory.js]: done.

**Comment by Chris-Hibbert** [comment]
@turadg, @dckc  Please take another look. A fair amount of work was required to bring this up-to-date. I'd like to merge it before the underlying code migrates again.

File: packages/inter-protocol/src/proposals/startPSM.js [hunk 7]
```diff
@@ -312,29 +312,6 @@ export const installGovAndPSMContracts = async ({
   );
 };
 
-/**
- * @deprecated the PSM charter has been merged with the econ charter
- * @param {EconomyBootstrapPowers} powers
- */
-export const startPSMCharter = async ({
```
Comment (dckc) [inline packages/inter-protocol/src/proposals/startPSM.js]: Isn't that still used by `packages/vats/src/core/boot-psm.js`? Ah... not as of fa9c030d27839d

**Review by dckc — APPROVED** [review]
yes, this looks ready to go.

File: packages/inter-protocol/src/proposals/econ-behaviors.js [hunk 8]
```diff
@@ -47,23 +48,41 @@ const MILLI = 1_000_000n;
...
  *   psmFacets: MapStore<Brand, PSMFacets>,
- *   econCharterStartResult: EconCharterStartResult,
- *   reservePublicFacet: import('../reserve/assetReserve.js').AssetReservePublicFacet,
- *   reserveCreatorFacet: import('../reserve/assetReserve.js').AssetReserveLimitedCreatorFacet,
- *   reserveGovernorCreatorFacet: GovernedAssetReserveFacetAccess,
- *   stakeFactoryCreatorFacet: StakeFactoryCreator,
- *   vaultFactoryCreator: VaultFactoryCreatorFacet,
- *   vaultFactoryGovernorCreator: GovernedContractFacetAccess<{},{}>,
- *   vaultFactoryVoteCreator: unknown,
+ *   econCharterFacets: EconCharterStartResult,
```
Comment (turadg) [inline packages/inter-protocol/src/proposals/econ-behaviors.js]: `SomeContractFacets` is the same as the results of its start function. In this PR,`someContractFacets` is used for the property at `SomeContractStartResult` for the type. Would it be better to be consistent between these cases?
```suggestion
 *   econCharterFacets: EconCharterFacets,
```

Or alternately,
```suggestion
 *   econCharterStartResult: EconCharterStartResult,
```

One argument against the latter is that it might read as the result of `startInstance` which would be inaccurate as that returns `StartedInstanceKit` which has `instance` and `adminFacet` too. Though that brings up another potential confusion: does `SomeContractFacets` include `adminFacet` or not? If not, then `SomeContractFacets` would be misleading.

This all isn't a blocker. Now seems like a good time to come up with consistent terminology for these types and bindings but it could be punted out to a ticket to finalize and implement.

File: packages/inter-protocol/src/proposals/econ-behaviors.js [hunk 9]
```diff
@@ -268,16 +284,17 @@ export const setupReserve = async ({
...
       poserInvitationP,
       E(E(zoe).getInvitationIssuer()).getAmountOf(poserInvitationP),
       feeMintAccessP,
+      ammFacetsP,
     ]);
 
   const reserveTerms = makeReserveTerms(
     poserInvitationAmount,
-    ammInstanceWithoutReserve,
+    ammFacets.instanceWithoutReserve,
```
Comment (turadg) [inline packages/inter-protocol/src/proposals/econ-behaviors.js]: this is confusing because `instance` isn't a facet. Maybe just call this `ammKit` to be less specific?

File: packages/inter-protocol/src/proposals/econ-behaviors.js [hunk 10]
```diff
@@ -647,43 +674,50 @@ export const startRewardDistributor = async ({
       return undefined;
     });
 
-  const feeDistributorFacets = await E(zoe).startInstance(
+  const facets = await E(zoe).startInstance(
```
Comment (turadg) [inline packages/inter-protocol/src/proposals/econ-behaviors.js]: please rename to match https://github.com/Agoric/agoric-sdk/blob/0c6bc9077a852d2a0b0bc6ee1c490274252e2cf3/packages/zoe/src/zoeService/utils.d.ts#L72-L77

```suggestion
  const instanceKit = await E(zoe).startInstance(
```

**Review by turadg — COMMENTED** [review]
I have some concerns around naming which lead me to not Approve it as is. I understand that this feature needs to get in and it's prone to merge conflicts so I'm not requesting changes, allowing Dan's extant approval to suffice.

File: packages/inter-protocol/src/proposals/econ-behaviors.js (hunk 8 above)
Comment (dckc) [reply packages/inter-protocol/src/proposals/econ-behaviors.js]: In at least some cases, the collection of facets is not exactly the result of `startInstance` but also includes, for example, a governor creator facet.

File: packages/inter-protocol/src/proposals/econ-behaviors.js (hunk 9 above)
Comment (dckc) [reply packages/inter-protocol/src/proposals/econ-behaviors.js]: true. `Kit` is also shorter (both in characters and in syllables) than `Facets`.

If it were a more widely exposed API, I might feel more strongly about it, but in this case, I don't mind much.

File: packages/inter-protocol/src/proposals/econ-behaviors.js (hunk 10 above)
Comment (Chris-Hibbert) [reply packages/inter-protocol/src/proposals/econ-behaviors.js]: done

File: packages/inter-protocol/src/proposals/econ-behaviors.js (hunk 8 above)
Comment (Chris-Hibbert) [reply packages/inter-protocol/src/proposals/econ-behaviors.js]: Yeah, It's definitely not the same as `*StartResult`, but it's also not composed of only things that are facets. I'm comfortable with `*Kit`. That naming convention doesn't seem to clash with other variable names.

File: packages/inter-protocol/src/proposals/econ-behaviors.js (hunk 9 above)
Comment (Chris-Hibbert) [reply packages/inter-protocol/src/proposals/econ-behaviors.js]: I'm convinced.

**Review by turadg — COMMENTED** [review]
👍
